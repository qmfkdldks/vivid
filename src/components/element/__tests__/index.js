import React from "react";
import { shallow } from "enzyme";
import Element from "../index";

// * it should evaluate element type and render corresponding element
// * it should contain children in element
// * when no element type was found, it renders `p` element

const defaultProps = { element: { type: "heading-two" } };

describe("Element", () => {
  it("renders successfully", () => {
    const wrapper = shallow(<Element {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders h2 when heading-two is given", () => {
    const wrapper = shallow(<Element {...defaultProps} />);
    expect(wrapper.find("h2").length).toEqual(1);
  });
});
