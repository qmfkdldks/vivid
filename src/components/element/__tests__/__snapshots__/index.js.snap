// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element should render successfully 1`] = `<h2 />`;
